src/control/cmd/dmg/auto_test.go

```diff
@@ -0,0 +1,92 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/lib/control"
+)
+
+func TestDmg_ConfigCommands(t *testing.T) {
+	runCmdTests(t, []cmdTest{
+		{
+			"Generate",
+			"config generate",
+			strings.Join([]string{
+				printRequest(t, &control.StorageScanReq{}),
```

Comment:
Shouldn't these be control.ConfigGenerateReqs? And contain the default values, or any params passed in?

Reply:
ConfigGenerateReq doesn't issue any RPCs but it does call Network/StorageScan which is what gets recorded here, these tests are simply checking that the parameters get accepted or not (an artefact of including the business logic inside the control API as requested)

Reply:
Ah okay. Makes sense.